integration/shareddata_test.go

```diff
@@ -25,23 +25,41 @@ package integration
 import (
 	"testing"
 
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+
 	"github.com/FerretDB/FerretDB/integration/setup"
 	"github.com/FerretDB/FerretDB/integration/shareddata"
 )
 
 func TestEnvData(t *testing.T) {
 	t.Parallel()
 
+	var previousName string
+
 	for _, p := range shareddata.AllProviders() {
-		p := p
 		name := p.Name()
 
 		t.Run(name, func(t *testing.T) {
-			setup.SetupWithOpts(t, &setup.SetupOpts{
+			s := setup.SetupWithOpts(t, &setup.SetupOpts{
 				DatabaseName:   "test",
-				CollectionName: p.Name(),
+				CollectionName: name,
 				Providers:      []shareddata.Provider{p},
 			})
+
+			// envData must persist, ensure the documents from the previous provider are found
```

Comment:
I'm not sure I understand the idea. 
In which case this might fail? I mean, in which case the documents won't be found here?

Reply:
This test would fail if you run this on `main`, because a temporary postgresql database is created for each test and dropped upon t.Cleanup, so previous collection no longer exists.

What is needed for `env-data` is not creating temporary database, but using `ferretdb` postgresql database and populating it with this test. Does this make sense?

Reply:
Updated the comment :hugs: